crates/release_plz_core/src/git/backend.rs

```diff
@@ -540,49 +563,163 @@ impl GitClient {
         };
 
         info!("opened pr: {}", git_pr.html_url);
-        self.add_labels(pr, git_pr.number)
+        self.add_labels(&pr.labels, git_pr.number)
             .await
             .context("Failed to add labels")?;
         Ok(git_pr)
     }
 
-    #[instrument(skip(self, pr))]
-    async fn add_labels(&self, pr: &Pr, pr_number: u64) -> anyhow::Result<()> {
-        if pr.labels.is_empty() {
+    #[instrument(skip(self))]
+    pub async fn add_labels(&self, labels: &[String], pr_number: u64) -> anyhow::Result<()> {
+        if labels.is_empty() {
             return Ok(());
         }
+
         match self.backend {
-            BackendType::Github => {
-                self.client
-                    .post(format!("{}/{}/labels", self.issues_url(), pr_number))
-                    .json(&json!({
-                        "labels": pr.labels
-                    }))
-                    .send()
-                    .await?
-                    .successful_status()
+            BackendType::Github => self.post_github_labels(labels, pr_number).await,
+            BackendType::Gitlab => self.post_gitlab_labels(labels, pr_number).await,
+            BackendType::Gitea => {
+                let (labels_to_create, mut label_ids) = self
+                    .get_pr_info_and_categorize_labels(pr_number, labels)
```

Comment:
here you assume that if a label is not present in PR number `pr_number` we need to create the label.

But what if the label already exists? Does Gitea return an error in this case?
If yes, shouldn't we check if the label already exist via their API?

For example, the user could have created this label for other purposes and they want to reuse it for the first time for the release PR

Reply:
https://docs.github.com/en/rest/issues/labels?apiVersion=2022-11-28#list-labels-for-a-repository

you could try this (maybe it works for gitea too). I haven't found the gitea docs in their swagger

Reply:
I see your point and i am going to change to
- getting all labels at the repository level not Pull request 
- this should solve for situations where ```For example, the user could have created this label for other purposes and they want to reuse it for the first time for the release PR```. By the time we are creating new labels it for the repository. 
Also, avoid duplicate labels.